activesupport/lib/active_support/log_subscriber/test_helper.rb

```diff
@@ -72,7 +72,7 @@ def method_missing(level, message = nil)
         end
 
         def logged(level)
-          @logged[level].compact.map { |l| l.to_s.strip }
+          @logged[level].filter_map { |l| l.to_s.strip }
```

Comment:
Is this rewriting safe?
```
irb(main):001:0> [nil].filter_map { |l| l.to_s.strip }
=> [""]
irb(main):002:0> [nil].compact.map { |l| l.to_s.strip }
=> []
```

Reply:
@yskkin Yes in my opinion, Because this code is a part of test helper written in `TestHelper::MockLogger`
no one will pass nil to test this out. 
Also When I checked in rails repository it is always used after a new initialization. And on initialization it will always make a clean array in @logged varialbe
`@logged = Hash.new { |h, k| h[k] = [] }`